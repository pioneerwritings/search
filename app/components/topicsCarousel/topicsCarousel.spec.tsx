import { render } from "~/test/test-utils";
import userEvent from '@testing-library/user-event'
import { type TopicsCarouselProps, TopicsCarousel } from './'

const props: TopicsCarouselProps = {
  data: ['Exercise', 'Nutrition', 'Water', 'Sunshine', 'Rest', 'Temperance', 'Trust In God', 'Fresh Air'],
  onClick: jest.fn(),
  activeItem: 'Exercise'
}

const renderComponent = (args?: TopicsCarouselProps) => {
  
  const { container, ...rest } = render(
    <TopicsCarousel {...args} {...props} />
  )
  
  return {
    container,
    topics: rest.getByRole('list'),
    ...rest
  }
}

describe('Topic Carousel', () => {
  it('Should render', () => {
    const { topics } = renderComponent()
    expect(topics).toBeInTheDocument()
  })

  it('Click handler should fire', async () => {
    const { getByRole } = renderComponent()
    const selected = getByRole('listitem', { name: 'Water' })

    await userEvent.click(selected)
    expect(props.onClick).toHaveBeenCalledWith('Water')
  })

  it('Matches the snapshot', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})