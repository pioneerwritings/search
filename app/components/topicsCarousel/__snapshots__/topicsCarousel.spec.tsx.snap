// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic Carousel Matches the snapshot 1`] = `
<div>
  <div
    class="
    topics-carousel 
    flex 
    items-center
    mx-auto 
    md:max-w-3xl 
    lg:max-w-3xl 
    xl:max-w-6xl
    mb-6
    md:mb-8
  "
  >
    <button
      aria-label="Click to scroll left"
      class="mr-4 shrink-0 rounded-full border items-center justify-center w-12 h-12 focus-visible:ring-2 focus-visible:ring-offset-4 focus-visible:outline-indigo focus-visible:ring-cornflower/50 hidden lg:flex"
      type="button"
    >
      <img
        src="/images/left-arrow.svg"
      />
    </button>
    <ul
      class="
    flex
    items-center
    overflow-x-auto 
    h-16 
    scrollbar-thin
    padding-0
    scroll-smooth
  "
      role="list"
    >
      <li
        aria-label="All"
        class="
    bg-gray-50
    h-10
    flex
    justify-center
    items-center
    pb-[2px]
    px-4
    text-md 
    cursor-pointer 
    rounded-full 
    text-center 
    font-bold 
    shrink-0
    mr-4
    hover:bg-gray-100
    focus:bg-gray-100
    focus:outline-black
    border
    hover:border-black
    transition-all
    list-none
   
    text-black
  "
        role="listitem"
        tabindex="0"
      >
        All
      </li>
      <li
        aria-label="Exercise"
        class="
    bg-gray-50
    h-10
    flex
    justify-center
    items-center
    pb-[2px]
    px-4
    text-md 
    cursor-pointer 
    rounded-full 
    text-center 
    font-bold 
    shrink-0
    mr-4
    hover:bg-gray-100
    focus:bg-gray-100
    focus:outline-black
    border
    hover:border-black
    transition-all
    list-none
   
    !bg-black
    text-white
    border-black
  "
        role="listitem"
        tabindex="0"
      >
        Exercise
      </li>
      <li
        aria-label="Nutrition"
        class="
    bg-gray-50
    h-10
    flex
    justify-center
    items-center
    pb-[2px]
    px-4
    text-md 
    cursor-pointer 
    rounded-full 
    text-center 
    font-bold 
    shrink-0
    mr-4
    hover:bg-gray-100
    focus:bg-gray-100
    focus:outline-black
    border
    hover:border-black
    transition-all
    list-none
   
    text-black
  "
        role="listitem"
        tabindex="0"
      >
        Nutrition
      </li>
      <li
        aria-label="Water"
        class="
    bg-gray-50
    h-10
    flex
    justify-center
    items-center
    pb-[2px]
    px-4
    text-md 
    cursor-pointer 
    rounded-full 
    text-center 
    font-bold 
    shrink-0
    mr-4
    hover:bg-gray-100
    focus:bg-gray-100
    focus:outline-black
    border
    hover:border-black
    transition-all
    list-none
   
    text-black
  "
        role="listitem"
        tabindex="0"
      >
        Water
      </li>
      <li
        aria-label="Sunshine"
        class="
    bg-gray-50
    h-10
    flex
    justify-center
    items-center
    pb-[2px]
    px-4
    text-md 
    cursor-pointer 
    rounded-full 
    text-center 
    font-bold 
    shrink-0
    mr-4
    hover:bg-gray-100
    focus:bg-gray-100
    focus:outline-black
    border
    hover:border-black
    transition-all
    list-none
   
    text-black
  "
        role="listitem"
        tabindex="0"
      >
        Sunshine
      </li>
      <li
        aria-label="Rest"
        class="
    bg-gray-50
    h-10
    flex
    justify-center
    items-center
    pb-[2px]
    px-4
    text-md 
    cursor-pointer 
    rounded-full 
    text-center 
    font-bold 
    shrink-0
    mr-4
    hover:bg-gray-100
    focus:bg-gray-100
    focus:outline-black
    border
    hover:border-black
    transition-all
    list-none
   
    text-black
  "
        role="listitem"
        tabindex="0"
      >
        Rest
      </li>
      <li
        aria-label="Temperance"
        class="
    bg-gray-50
    h-10
    flex
    justify-center
    items-center
    pb-[2px]
    px-4
    text-md 
    cursor-pointer 
    rounded-full 
    text-center 
    font-bold 
    shrink-0
    mr-4
    hover:bg-gray-100
    focus:bg-gray-100
    focus:outline-black
    border
    hover:border-black
    transition-all
    list-none
   
    text-black
  "
        role="listitem"
        tabindex="0"
      >
        Temperance
      </li>
      <li
        aria-label="Trust In God"
        class="
    bg-gray-50
    h-10
    flex
    justify-center
    items-center
    pb-[2px]
    px-4
    text-md 
    cursor-pointer 
    rounded-full 
    text-center 
    font-bold 
    shrink-0
    mr-4
    hover:bg-gray-100
    focus:bg-gray-100
    focus:outline-black
    border
    hover:border-black
    transition-all
    list-none
   
    text-black
  "
        role="listitem"
        tabindex="0"
      >
        Trust In God
      </li>
      <li
        aria-label="Fresh Air"
        class="
    bg-gray-50
    h-10
    flex
    justify-center
    items-center
    pb-[2px]
    px-4
    text-md 
    cursor-pointer 
    rounded-full 
    text-center 
    font-bold 
    shrink-0
    mr-4
    hover:bg-gray-100
    focus:bg-gray-100
    focus:outline-black
    border
    hover:border-black
    transition-all
    list-none
   
    text-black
  "
        role="listitem"
        tabindex="0"
      >
        Fresh Air
      </li>
    </ul>
    <button
      aria-label="Click to scroll right"
      class="ml-4 shrink-0 rounded-full border items-center justify-center w-12 h-12 focus-visible:ring-2 focus-visible:ring-offset-4 focus-visible:outline-indigo focus-visible:ring-cornflower/50 hidden lg:flex"
      type="button"
    >
      <img
        src="/images/right-arrow.svg"
      />
    </button>
  </div>
</div>
`;
